src/h5web/providers/Provider.tsx

```diff
@@ -18,8 +20,28 @@ function Provider(props: Props): ReactElement {
     return api.getMetadata();
   }, [api]);
 
-  const groupsStore = useMemo(() => {
-    const store = createFetchStore(api.getGroup.bind(api));
+  const entitiesStore = useMemo(() => {
+    const childCache = new Map<string, Entity>();
+
+    const store = createFetchStore(async (path: string) => {
+      if (childCache.has(path)) {
+        return childCache.get(path) as Entity;
+      }
+
+      const entity = await api.getEntity(path);
+
+      if (isGroup(entity)) {
+        // Cache non-group children (datasets, datatypes and links)
+        entity.children.forEach((child) => {
+          if (!isGroup(child)) {
+            childCache.set(path, entity);
+          }
+        });
+      }
+
+      return entity;
+    });
```

Comment:
Even if in the end there is a single `entitiesStore`, to build it, you maintain two separate cache: `childCache` and the one from `store`.

Do you think it might be possible to use the `store` API to add directly the children to it and avoid keeping a separate cache ?

Reply:
The `store` API is pretty limited: https://github.com/dai-shi/react-suspense-fetch/blob/master/src/index.ts#L12-L16 -- it doesn't provide access to its internal cache.